features/329-principal-apis/api.md

```diff
@@ -0,0 +1,270 @@
+# Design Summary
+
+## SDK API
+```typescript
+type Metadata = { [key: string]: unknown };
+
+interface User {
+    id: string;
+    createdAt: Date;
+    lastLoginAt: Date;
+    isVerified: boolean;
+    metadata: Metadata;
+
+    identity: Identity;
+}
+
+interface IdentityBase {
+    type: string;
+    metadata: Metadata;
+}
+interface PasswordIdentity extends IdentityBase {
+    type: 'password';
+    loginIDKey: string;
+    loginID: string;
+    realm: string;
+}
+interface OAuthIdentity extends IdentityBase {
+    type: 'oauth';
+    providerID: string;
+    providerUserID: string;
+    metadata: {
+        raw_profile: object;
+    };
+}
+interface CustomTokenIdentity extends IdentityBase {
+    type: 'custom';
+    providerUserID: string;
+    metadata: {
+        raw_profile: object;
+    };
+}
+type Identity = PasswordIdentity | OAuthIdentity | CustomTokenIdentity;
```

Comment:
- Remove `IdentityBase`
- Move `raw_profile` to top-level
- For OAuth `metadata: { email?: string }`
- For Custom Token `metadata: { email?: string }`
- For Password, `metadata: { email?: string, phone?: string }`

Specify `metadata` in Identity is always derived. Specify how it is derived for each Identity.